types/acriosSystems/ACR-CV-101L-M-X/uplink.js

```diff
@@ -0,0 +1,730 @@
+		}
+	} else {
+		if(b == 5) {
+			--b;
+			m = ba2f;
+		} else {
+			if(b == 7) ++b;
+			m = t&8 ? ba2bcd : ba2i;
+		}
+	}
+	i = t = sl("Record #" + v.id, b);
+	if(m) {
+		try {
+			t = m(t);
+		} catch(e) {
+			v.error = true;
+			t = e;
+		}
+	}
+	if(!v.error) {
+		if(v.f) t = v.f(t);
+		m = Array.isArray(t);
+		if(m) t = ha2si(t);
+		if(s) t = m ? (t[0] == "-" ? t.slice(1) : ("-" + t)): -t;
+		if(v.e) t = p10(t, v.e);
+	}
+	v.value = t;
+	v.rawValue = i;
+	v.func = vFunc[f];
+	d >>= 6; f = d&1; i = t = u = 0;
+	if(d&2) {
+		for(; i < l; ++i) {
+			d = y[i + 1];
+			u += (d>>6&1)<<i;
+			t += (d>>4&3)<<i*2;
+			f += (d&0xF)<<i*4+1;
+		}
+		v.device = u;
+		v.tariff = t;
+	}
+	v.storage = f;
+	delete v.f;
+	delete v.e;
+}
+
+function pV() {
+	r.manId = deManId(a, n);
+	r.version = i(), i();
+	sD(c, deD(c));
+	r.accessN = i();
+	r.status = i();
+	deS(r);
+	n += 2;
+	while(n < e - 1) {
+		var t = a[n], v = t == 0x2F ? v : nv();
+		if((t&0xF) == 0xF) {
+			t = t>>4&7; ++n;
+			if(t < 2) {
+				if(t) v.request = "Readout again";
+				v.type = MS;
+				v.value = sl(v.type, e - n);
+			} else if(t > 6) v.request = "Global readout";
+		} else {
+			v.dif = rif([]);
+			v.vif = rif([]);
+			rv(v);
+		}
+	}
+}
+
+r.fixed ? pF() : pV();
+return r;
+}
+
+function unitConv(cfg, f) {
+
+var U = ["J", "Wh", "W", "J/h"], P = ["", "k", "M", "G"], k, g;
+
+function x(d) {
+	var u = d.unit, v = d.value, m;
+	if(u && v) {
+		if(m = k[u]) try {
+			d.value = p10(v, m[0]);
+			d.unit = m[1];
+		} catch(e) {}
+	}
+	if(g) try {
+		g(d);
+	} catch(e) {}
+}
+
+function p(d) {
+	var a, i;
+	if(d && (a = d.data)) for(i in a) x(a[i]);
+	return d;
+}
+
+function c(d, f) {
+	k = {}, g = f;
+	var u, c, o, n, v;
+	if(d) for(o in d) {
+		c = d[o], u = U.indexOf(o), v = P.indexOf(c);
+		if(u >= 0 && v >= 0) for(n in P) k[P[n] + o] = [(n - v) * 3, c + o];
+	}
+}
+
+c(cfg, f);
+
+return {
+	getUnits: function() {return U.slice();},
+	getPrefixes: function() {return P.slice();},
+	config: c,
+	process: p
+};
+
+}
+
+
+/**
+* The consume(event) function is the entry point for the script and will be invoked upon execution.
+* An error will be returned if the script doesn't implement a consume(event) function.
+* @param {ConsumeEvent} event
+*/ 
+function consume(event) {
+	var lastFrameIndex = (event.state || {}).lastFrameIndex;
+	var thisFrameHex = event.data["payloadHex"];
+	var thisFrameIndex = parseInt(thisFrameHex.substr(0,2),16);
+	var totalBatchFrames = parseInt(thisFrameHex.substr(2,2),16);
+	var thisFramePayload = thisFrameHex.substr(4);
+	var previousPayloads = (event.state || {}).previousPayloads;
+	var data;
+
+	if (typeof lastFrameIndex === "undefined") {
+		lastFrameIndex = 0;
+	}
+
+	if((lastFrameIndex+1) != thisFrameIndex) {
+		if(thisFrameIndex != 1) {
+			emit("state", {lastFrameIndex: 0, previousPayloads: []});
+			return;
+		}
+		previousPayloads = []
+	}
+
+	if(thisFrameIndex == totalBatchFrames){
+		if(thisFrameIndex == 1) {
+			data = tmbus(thisFramePayload);
```

Comment:
likely this needs to be changed to, as data needs to be an object

`data = JSON.parse(tmbus(thisFramePayload));`

Reply:
Hi @mariusgiger,

Thanks for pointing that out. The issue was that the parser was also emitting state and was still validating against the deleted schema. I forgot to modify this part when I removed the schema. I've made the necessary adjustments, and the test is now passing.